python/dask_cudf/dask_cudf/groupby.py

```diff
@@ -255,7 +255,12 @@ def groupby_agg(
         # be str, rather than tuples).
         for col in aggs:
             _aggs[col] = _aggs[col][0]
-    _meta = ddf._meta.groupby(gb_cols, as_index=as_index).agg(_aggs)
+    try:
+        _meta = ddf._meta.groupby(gb_cols, as_index=as_index).agg(_aggs)
+    except NotImplementedError:
+        _meta = ddf._meta_nonempty.groupby(gb_cols, as_index=as_index).agg(
+            _aggs
+        )
```

Comment:
Did the `list` aggregation not work on an empty dataframe here or return a different type than expected or something?

Reply:
This also occurred on non-empty dataframes; here's the traceback:

```
Traceback (most recent call last):
  File "cudf/_lib/groupby.pyx", line 164, in cudf._lib.groupby.GroupBy.aggregate
RuntimeError: cuDF failure at: /raid/charlesb/github/cudf/cpp/src/column/column_factories.cpp:68: make_empty_column is invalid to call on nested types

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "python/dask_cudf/test_dask_collect.py", line 21, in <module>
    print(gddf.groupby("a").agg({"b":list}).compute())
  File "/raid/charlesb/github/dask/dask/dataframe/groupby.py", line 1847, in agg
    return self.aggregate(arg, split_every=split_every, split_out=split_out)
  File "/raid/charlesb/github/cudf/python/dask_cudf/dask_cudf/groupby.py", line 70, in aggregate
    return groupby_agg(
  File "/raid/charlesb/github/cudf/python/dask_cudf/dask_cudf/groupby.py", line 259, in groupby_agg
    _meta = ddf._meta.groupby(gb_cols, as_index=as_index).agg(_aggs)
  File "/raid/charlesb/miniconda3/envs/cudf_dev/lib/python3.8/contextlib.py", line 75, in inner
    return func(*args, **kwds)
  File "/raid/charlesb/miniconda3/envs/cudf_dev/lib/python3.8/site-packages/cudf-0.20.0a0+170.gb0218b45e7.dirty-py3.8-linux-x86_64.egg/cudf/core/groupby/groupby.py", line 164, in agg
    result = self._groupby.aggregate(self.obj, normalized_aggs)
  File "cudf/_lib/groupby.pyx", line 173, in cudf._lib.groupby.GroupBy.aggregate
NotImplementedError: Aggregation not supported for empty columns
```

Which was happening when trying to do:

https://github.com/rapidsai/cudf/blob/32222de53b6f97fb26f01ee4e413c72019c0102c/python/cudf/cudf/_lib/groupby.pyx#L171-L175

Reply:
cc @shwina do you happen to know if this is a Python thing or a libcudf thing?

Reply:
It seems like the underlying blocker is `libcudf`, specifically #7611. There appears to be some confusion on the nature of the problem in #7611, perhaps @mythrocks can provide some context as to whether this is actually something Python `cudf` needs to be disabling or if the issue raised was erroneous and we would be safe to reenable this in Python.

Reply:
I was hoping to address the rolling-window side of #7611 some time this week, if all goes well.

Do I understand correctly that you're running into this in the group-by case, for empty input?

Reply:
Just to close the loop here, Mithun and I chatted offline to understand where the underlying problem is. It's definitely in `libcudf`, not `cudf` Python. Fortunately it looks like `dask_cudf` is capable of handling this for the moment, so hopefully it's not a blocker.

Reply:
Just for clarification, this case happens for non-empty input - in particular, the example above:

```python
data = """a,b
    1595802,1611:0.92
    1595802,1610:0.07
    1524246,1807:0.92
    1524246,1608:0.07"""

df = pd.read_csv(StringIO(data))
ddf = dd.from_pandas(df, 2)

gdf = cudf.from_pandas(df)
gddf = dask_cudf.from_cudf(gdf, 2)

print(ddf.groupby("a").agg({"b":list}).compute())
print(gddf.groupby("a").agg({"b":list}).compute())
```

Reply:
Hello, @charlesbluca. Would it be viable to retest your case with the fix I have up on #8279?

Reply:
P.S. Sorry for the wait.

Reply:
Can verify that #8279 resolves this! Thanks @mythrocks 😃